main/src/hadd.cxx

```diff
@@ -427,10 +427,10 @@ int main( int argc, char **argv )
          else
             newcomp = ROOT::RCompressionSetting::EDefaults::kUseCompiledDefault;
          delete firstInput;
-         fileMerger.SetMergeOptions(TString("first_source_compression"));
+         fileMerger.SetMergeOptions(TString("rnt:FirstSrcCompression"));
```

Comment:
Why does it need to be handled differently than the 'same' option for `TTree`?

Reply:
I'm not sure how TTree handles the case, but my reasoning is that, since RNTuple has a different default compression than TFile/TTree, we want to distinguish the cases of "use default compression" (user didn't pass any flag to hadd) and that of "use first file compression" (user passed -ff or -fk).
As far as I know, we have no way of telling which case are we in from RNTuple::Merge, so we rely on explicit merge options for that.

Do you think there is a better way of handling it? I might be missing something that would let us avoid these options.

Reply:
In this case, there is no good reason to make this option being 'RNTuple' only.  The fact that `TTree` is currently cumbersomely doing an approximate implementation is interesting but not relevant (IMHO).

Currently `TTree` uses the first file's global compression setting to set the compression to use either:
* This is the right choice, then for symmetry reason it should be the same choice for RNTuple (and if RNTuple wants/needs more, we may need a new option 'use the compression of the first RNTuple/TTree').
* This is not the right choice, then TTree should be updated to switch behavior when seeing the flag to be like RNTUple and use the compression of the first TTree.